packages/mui-material/src/Grid/Grid.js

```diff
@@ -29,71 +29,80 @@ function getOffset(val) {
   return `${parse}${String(val).replace(String(parse), '') || 'px'}`;
 }
 
-function generateGrid(globalStyles, theme, breakpoint, ownerState) {
-  const size = ownerState[breakpoint];
-
-  if (!size) {
-    return;
-  }
-
-  let styles = {};
-
-  if (size === true) {
-    // For the auto layouting
-    styles = {
-      flexBasis: 0,
-      flexGrow: 1,
-      maxWidth: '100%',
-    };
-  } else if (size === 'auto') {
-    styles = {
-      flexBasis: 'auto',
-      flexGrow: 0,
-      flexShrink: 0,
-      maxWidth: 'none',
-      width: 'auto',
-    };
-  } else {
-    const columnsBreakpointValues = resolveBreakpointValues({
-      values: ownerState.columns,
-      breakpoints: theme.breakpoints.values,
-    });
-
-    const columnValue =
-      typeof columnsBreakpointValues === 'object'
-        ? columnsBreakpointValues[breakpoint]
-        : columnsBreakpointValues;
-    // Keep 7 significant numbers.
-    const width = `${Math.round((size / columnValue) * 10e7) / 10e5}%`;
-    let more = {};
+function generateGrid({ theme, ownerState }) {
+  let size;
+  const breakpointBase = theme.breakpoints.keys.reduce((breakpoints, breakpoint) => {
```

Comment:
It took me awhile to realize that resolveBreakpointValues has an optional parameter that specifies what breakpoints are outputted.  I calculate this here so we don't have to redo it for every breakpoint.

Reply:
I didn't see an easier way in breakpoints.js to create `{ xs: true, sm: true, md: true, lg: true, xl: true }`